bb8/src/inner.rs

```diff
@@ -148,7 +158,9 @@ where
 
     /// Returns information about the current state of the pool.
     pub(crate) fn state(&self) -> State {
-        (&*self.inner.internals.lock()).into()
+        let mut state: State = (&*self.inner.internals.lock()).into();
+        state.statistics = (&(self.inner.statistics)).into();
+        state
```

Comment:
I do very dislike very much what i had to here 😓 

So either you know how to do that this better, if not my porposal if we wan to keep this aggregated is to just use three different object, `State`, `PoolState` and `Statistics`. The `State` one would have the instances of the `PoolState` and `Statistics`.

Thoughts @djc ?

Reply:
@djc pinging you kere for when yo have a moment 🙇

Reply:
Took the liberty of pushing to your branch. Let's see what CI thinks.